rails/app/models/theme.rb

```diff
@@ -9,9 +9,18 @@ class Theme < ApplicationRecord
   validates :sponsor_logos, blob: { content_type: ['image/png', 'image/jpg', 'image/jpeg'], size_range: 1..5.megabytes }
   validates :mapbox_access_token, :presence =>  {:message => 'is required when the Mapbox style URL is set.'}, unless: -> { mapbox_style_url.blank? }
   validates :mapbox_style_url, :presence => {:message => 'is required when the Mapbox access token is set.'}, unless: -> { mapbox_access_token.blank? }
-  validates :center_lat, :sw_boundary_lat, :ne_boundary_lat,
+  if :sw_boundary_long == nil and :sw_boundary_lat == nil and :ne_boundary_lat == nil and :ne_boundary_long == nil
+    validates :sw_boundary_long, :ne_boundary_long, :sw_boundary_lat, :ne_boundary_lat,
+      absence: true # update to remove bounding box if all bb coordinates are empty, else enforce constraints
+  else
+    validates :sw_boundary_lat, :ne_boundary_lat,
+        :numericality=> true, allow_nil: true, :inclusion => {:in => -90..90, :message => "value should be between -90 and 90"}
+    validates :sw_boundary_long, :ne_boundary_long,
+        :numericality=> true, allow_nil: true, :inclusion => {:in => -180..180, :message => "value should be between -180 and 180"}
+  end
```

Comment:
I don't think this will work, since (I believe) the `if/else` statements will be evaluated when the model is loaded (that is, on app initialization) rather than when the model is validated. That means the `else` branch will always be active (since `:sw_boundary_long == nil` will _always_ be false) 

However you can use [conditional validation](https://guides.rubyonrails.org/active_record_validations.html#conditional-validation). Alternately you could write a [custom validator class](https://guides.rubyonrails.org/active_record_validations.html#performing-custom-validations) that checks everything, but given what you've already written the first way is probably easier:
```suggestion
  validates :sw_boundary_long, :ne_boundary_long, :sw_boundary_lat, :ne_boundary_lat,
    absence: true,
    if: :nil_boundaries?
    # update to remove bounding box if all bb coordinates are empty, else enforce constraints
  validates :sw_boundary_lat, :ne_boundary_lat,
      numericality: true,
      allow_nil: true,
      inclusion: {in: -90..90, message: "value should be between -90 and 90"},
      unless: :nil_boundaries?
  validates :sw_boundary_long, :ne_boundary_long,
      numericality: true,
      allow_nil: true,
      inclusion: {in: -180..180, message: "value should be between -180 and 180"},
      unless: :nil_boundaries?
      
    def nil_boundaries?
      sw_boundary_long.nil? and sw_boundary_lat.nil? and ne_boundary_lat.nil? and ne_boundary_long.nil?
    end
```

Reply:
Thinking a little bit more about this, I think you can get away with only the last 2 validations with the conditional, and change `allow_nil` to `false`. This should satisfy what you mentioned in bullet 2 (either all are nil, or none are and are numerical within the proper range - am I reading that right?) The first validation would just be redundant (validate all 4 values are nil if all four values are already nil.)
```suggestion
  validates :sw_boundary_lat, :ne_boundary_lat,
    numericality: true,
    allow_nil: false,
    inclusion: {in: -90..90, message: "value should be between -90 and 90"},
    unless: :nil_boundaries?
  validates :sw_boundary_long, :ne_boundary_long,
    numericality: true,
    allow_nil: false,
    inclusion: {in: -180..180, message: "value should be between -180 and 180"},
    unless: :nil_boundaries?
    
  def nil_boundaries?
    sw_boundary_long.nil? and sw_boundary_lat.nil? and ne_boundary_lat.nil? and ne_boundary_long.nil?
  end
```

Reply:
> I don't think this will work, since (I believe) the if/else statements will be evaluated when the model is loaded (that is, on app initialization) rather than when the model is validated. That means the else branch will always be active (since :sw_boundary_long == nil will always be false)

I believe even if the entire conditional was moved into a conditional validation and loaded at call, it would still always be false. Symbols defined as they are here are values *as themselves* and so would never be nil.

e.g.

```ruby
:test
=> :test
:test.nil?
=> false
```

I really like the adjusted validations that @viamin suggested with only using the two validations and disallowing nil unless all 4 values are nil.

We could also take it a step further and handle this in a [custom validation method](https://guides.rubyonrails.org/v6.0/active_record_validations.html#custom-methods) instead (not to be confused with a custom validator class). This would make it extremely obvious that the map bounds are all required, or all nil. The fact that we have two separate validations that then use the same `nil_boundaries?` check feels like this may be a warranted move.

Something akin to:

```suggestion
  validate :map_bounds
  
  def map_bounds
    return if sw_boundary_long.nil? && sw_boundary_lat.nil? # southwest bounds
       && ne_boundary_long.nil? && ne_boundary_lat.nil? # northeast bounds
    
    errors.add(:sw_boundary_lat, "value should be between -90 and 90") unless (-90..90).include?(sw_boundary_lat)
    errors.add(:ne_boundary_lat, "value should be between -90 and 90") unless (-90..90).include?(ne_boundary_lat)
    errors.add(:sw_boundary_long, "value should be between -180 and 180") unless (-180..180).include?(sw_boundary_long)
    errors.add(:ne_boundary_long, "value should be between -180 and 180") unless (-180..180).include?(ne_boundary_long)
  end
```